cmd/k3d/create.go

```diff
@@ -1187,34 +1187,39 @@ func runK3d(cmd *cobra.Command, _ []string) error {
 	progressPrinter.AddTracker("wrapping-up", "Wrapping up", 2)
 	progressPrinter.SetupProgress(progressPrinter.TotalOfTrackers(), false)
 
-	if err := k3d.PostRunPrepareGitopsRepository(config.GitopsDir); err != nil {
-		return fmt.Errorf("error detokenizing post run: %w", err)
-	}
+	executionControl = viper.GetBool("kubefirst-checks.post-detokenize")
+	if !executionControl {
+		if err := k3d.PostRunPrepareGitopsRepository(config.GitopsDir); err != nil {
+			return fmt.Errorf("error detokenizing post run: %w", err)
+		}
 
-	gitopsRepo, err := git.PlainOpen(config.GitopsDir)
-	if err != nil {
-		return fmt.Errorf("error opening repo at %q: %w", config.GitopsDir, err)
-	}
-	_, err = os.Stat(fmt.Sprintf("%s/terraform/%s/remote-backend.md", config.GitopsDir, config.GitProvider))
-	if err == nil {
-		err = os.Rename(fmt.Sprintf("%s/terraform/%s/remote-backend.md", config.GitopsDir, config.GitProvider), fmt.Sprintf("%s/terraform/%s/remote-backend.tf", config.GitopsDir, config.GitProvider))
+		gitopsRepo, err := git.PlainOpen(config.GitopsDir)
 		if err != nil {
-			return fmt.Errorf("failed to rename remote-backend.md to remote-backend.tf: %w", err)
+			return fmt.Errorf("error opening repo at %q: %w", config.GitopsDir, err)
+		}
+		_, err = os.Stat(fmt.Sprintf("%s/terraform/%s/remote-backend.md", config.GitopsDir, config.GitProvider))
+		if err == nil {
+			err = os.Rename(fmt.Sprintf("%s/terraform/%s/remote-backend.md", config.GitopsDir, config.GitProvider), fmt.Sprintf("%s/terraform/%s/remote-backend.tf", config.GitopsDir, config.GitProvider))
+			if err != nil {
+				return fmt.Errorf("failed to rename remote-backend.md to remote-backend.tf: %w", err)
+			}
 		}
```

Comment:
This seems like the opposite of what you want to do and it's considered an anti-pattern in Go: not only it eats the error, it also handles the non-error like an error.

If you can't stat the file, do you still continue?

Reply:
I didn't write this code, I only wrapped it in a condition so it doesn't run again for retries. Other than that I Agree

Reply:
I'll remove the Stat since we have talked a lot about this 😄